apps/wallet/src/ui/app/KeypairVault.ts

```diff
@@ -3,13 +3,17 @@
 
 import { Ed25519Keypair } from '@mysten/sui.js';
 
+import { toEntropy, entropyToMnemonic } from '_shared/utils/bip39';
+
 import type { Keypair } from '@mysten/sui.js';
 
 export default class KeypairVault {
     private _keypair: Keypair | null = null;
 
-    public set mnemonic(mnemonic: string) {
-        this._keypair = Ed25519Keypair.deriveKeypair(mnemonic);
+    public set entropy(entropy: string) {
+        this._keypair = Ed25519Keypair.deriveKeypair(
```

Comment:
As a follow-up, we should create a variant of this that allows keypair derivation from entropy instead of mnemonics.